### PR TITLE
JAMES-2856 Buggy quotaRoot serialization in mailbox event serialization

### PR DESCRIPTION
**Pull request by chibenwa**
Impact: On top of **guice-distributed** product, quota search 'domain criterion' was always empty thus not working.

### CODE REVIEW

**Comment by Arsnael** [comment]
Why opening this PR when I see the same fix being applied in https://github.com/linagora/james-project/pull/2587?

**Comment by chibenwa** [comment]
> Why opening this PR when I see the same fix being applied in #2587?

It's not that much related to the work there & the impact is wider (bug) hence I want to handle that separately. I'll ask @billyfrost418 to remove these 2 commits from his PR.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultUserQuotaRootResolver.java [hunk 1]
```diff
@@ -42,15 +43,35 @@
...
 
     private final SessionProvider sessionProvider;
     private final MailboxSessionMapperFactory factory;
+    private final DefaultQuotaRootDeserializer quotaRootDeserializer;
 
     @Inject
     public DefaultUserQuotaRootResolver(SessionProvider sessionProvider, MailboxSessionMapperFactory factory) {
         this.sessionProvider = sessionProvider;
         this.factory = factory;
+        quotaRootDeserializer = new DefaultQuotaRootDeserializer();
```
Comment (blackheaven) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/DefaultUserQuotaRootResolver.java]: can be `static`

**Comment by chibenwa** [comment]
test this please

Unit test OK

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please (ObjectStorageBlobsDAOTest.saveConcurrentlyWithNonPreExistingBucketShouldNotFail » NotTerminated)

**Comment by chibenwa** [comment]
test this please (deployment tests)

**Comment by chibenwa** [comment]
test this please (deployment tests)

**Comment by mbaechler** [comment]
Is the serialization for internal use or is it exposed?

**Comment by chibenwa** [comment]
internal use (event bus)

**Comment by chibenwa** [comment]
test this please (deployment tests)

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Just rebased ;-)

**Comment by chibenwa** [comment]
```
CassandraAccessTokenRepositoryTest>AccessTokenRepositoryTest.validTokenMustBeRetrieved:45 » InvalidAccessToken
```

test this please

**Pull request closed without merging**